camayoc/tests/qcs/api/v1/sources/test_sources_common.py

```diff
@@ -267,16 +267,18 @@ def test_read_all(src_type, cleanup, shared_client):
     :expectedresults: All sources are present in data returned by API.
     """
     created_srcs = []
-    for _ in range(random.randint(2, 20)):
+    for _ in range(random.randint(2, 10)):
         # gen_valid_srcs will take care of cleanup
         created_srcs.append(gen_valid_source(cleanup, src_type, 'localhost'))
-    server_srcs = Source().list().json()
-    for sp in server_srcs:
-        for i, cp in enumerate(created_srcs):
-            if sp['id'] == cp.fields()['id'] and cp.equivalent(sp):
-                created_srcs.remove(cp)
+    server_srcs = Source().list().json()['results']
+    num_srcs = len(created_srcs)
+    for server_source in server_srcs:
+        for created_source in created_srcs:
+            if server_source['id'] == created_source.fields()['id']:
+                assert_matches_server(created_source)
```

Comment:
Now will fail on the guilty source

Reply:
This is good but it will only catch the first "invalid" occurrence. What if multiple sources are "invalid"?

I am leaving this as something to think about. I mean we can go with this as the improvement for now, but as a future though I think we should start trying to provide all the problematic instances in the future.